Starbucks/Controllers/ProductsViewController/ProductsViewController.swift

```diff
@@ -8,10 +8,78 @@
 import Foundation
 import UIKit
 
-class ProductsViewController: UIViewController {
+class ProductsViewController: UIViewController, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
     
+    let spacing: CGFloat = 2
+    let products: [Product]
+    
+    lazy var productCollectionView: UICollectionView = {
+        let flowLayout = UICollectionViewFlowLayout()
+        flowLayout.scrollDirection = .vertical
+        flowLayout.minimumLineSpacing = spacing
+        flowLayout.minimumInteritemSpacing = spacing
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: flowLayout)
+        collectionView.isPagingEnabled = true
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        return collectionView
+    }()
+    
+    init(products: [Product]) {
+        self.products = products
+        super.init(nibName: nil, bundle: nil)
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        view.backgroundColor = .green
+        view.backgroundColor = UIColor.white
+        setupConstraints()
+        registerCollectionCells()
+    }
+    
+    func setupConstraints() {
+        view.addSubview(productCollectionView)
+        
+        productCollectionView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 20).isActive = true
+        productCollectionView.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -20).isActive = true
```

Comment:
You could alternatively use the `contentInset` to define these edges. Constraints are also fine.